### PR TITLE
Config validator

### PR DESCRIPTION
**Pull request by kevensen**
Initial commit of config-validator

Merge after #24

### CODE REVIEW

File: charts/forseti-security/values.yaml [hunk 1]
```diff
@@ -15,6 +15,15 @@
...
 cloudsqlConnection: ""
 
+# configValidator sets whether or not to deploy config-validator
+configValidator: true
+
+# configValidatorImage is the container image used by the config-validator
+configValidatorImage: gcr.io/forseti-containers/config-validator
+
+# configValidatorImageTag is the tag for the config-validator image.
+configValidatorImageTag: latest
```
Comment (hshin-g) [inline charts/forseti-security/values.yaml:25]: Does this need to be a string?

File: charts/forseti-security/templates/server/pod/_sql-container.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  - name: MYSQL_ALLOW_EMPTY_PASSWORD
+    value: "1"
+  - name: MYSQL_DATABASE
+    value: forseti_security
+  ports:
+  - containerPort: 3306
+  volumeMounts:
+  - name: mysql-disk
+    mountPath: /var/lib/mysql
+{{ end }}
```
Comment (hshin-g) [inline charts/forseti-security/templates/server/pod/_sql-container.yaml]: New line here

File: charts/forseti-security/values.yaml (hunk 1 above)
Comment (kevensen) [reply charts/forseti-security/values.yaml:25]: Actually, these values get interpreted as strings by default.

File: charts/forseti-security/templates/server/persistent-volume-claims/_policy-library.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+kind: PersistentVolumeClaim
+metadata:
+  name: policy-library
+spec:
+  accessModes:
+    - ReadWriteOnce
+    - ReadOnlyMany
+  resources:
+    requests:
+      storage: 1Gi
```
Comment (hshin-g) [inline charts/forseti-security/templates/server/persistent-volume-claims/_policy-library.yaml]: New line here

File: charts/forseti-security/templates/server/pod/_cloudsqlproxy-container.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  image: gcr.io/cloudsql-docker/gce-proxy:latest
+  command: ["/cloud_sql_proxy", "-instances={{ .Values.cloudsqlConnection }}=tcp:3306", "-credential_file=/var/secrets/google/key.json"]
+  securityContext:
+    runAsUser: 2 
+    allowPrivilegeEscalation: false
+  volumeMounts:
+  - name: google-cloud-key
+    mountPath: /var/secrets/google
+    readOnly: true
+{{ end }}
```
Comment (hshin-g) [inline charts/forseti-security/templates/server/pod/_cloudsqlproxy-container.yaml]: New line here

File: charts/forseti-security/templates/server/pod/_sql-container.yaml (hunk 2 above)
Comment (kevensen) [reply charts/forseti-security/templates/server/pod/_sql-container.yaml]: yup, will fix

File: charts/forseti-security/templates/server/pod/_policy_library_importer.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  image: gcr.io/cloud-builders/git:latest
+  args: ["clone", "{{ .Values.policyLibraryRepositoryURL }}", "/policy-library/"]
+  securityContext:
+    runAsUser: 65532
+    runAsGroup: 65532
+    fsGroup: 65532
+  volumeMounts:
+  - name: policy-library
+    mountPath: /policy-library
+{{ end }}
```
Comment (hshin-g) [inline charts/forseti-security/templates/server/pod/_policy_library_importer.yaml]: New line here

File: charts/forseti-security/templates/server/pod/_server-container.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      command:
+      - forseti
+      - inventory
+      - list
+    initialDelaySeconds: 30
+    periodSeconds: 30
+    timeoutSeconds: 5
+    successThreshold: 1
+    failureThreshold: 3
+{{ end }}
```
Comment (hshin-g) [inline charts/forseti-security/templates/server/pod/_server-container.yaml]: New line here

File: charts/forseti-security/templates/config-validator/secrets/git_sync_ssh_key.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+{{- if and (.Values.gitSyncSSH) (.Values.configValidator) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: git-sync-ssh-key
+type: Opaque
+data:
+  ssh: >
+    {{ .Values.gitSyncPrivateSSHKey }}
+{{- end }}
```
Comment (hshin-g) [inline charts/forseti-security/templates/config-validator/secrets/git_sync_ssh_key.yaml]: Newline

**Review by hshin-g — APPROVED** [review]

**Pull request merged**